### PR TITLE
Improve handling of snapshot path and fix bugs with it

### PR DESCRIPTION
**Pull request by Blaisorblade**
Split out from #430. Gonna merge based on the existing review there.

### CODE REVIEW

File: app/src/main/java/com/kamron/pogoiv/Pokefly.java [hunk 1]
```diff
@@ -255,7 +255,7 @@
     private Optional<Integer> pokemonCP = Optional.absent();
     private Optional<Integer> pokemonHP = Optional.absent();
     private double estimatedPokemonLevel = 1.0;
-    private @NonNull Optional<String> ssFile = Optional.absent();
+    private @NonNull Optional<String> screenShotPath = Optional.absent();
```
Comment (sarav) [inline app/src/main/java/com/kamron/pogoiv/Pokefly.java]: Path is ambiguous -- whether it's dir or file. That's why I had used File. But not a big deal.

**Comment by sarav** [comment]
I'm still getting a crash on isEmpty() in deleteIfRequired() when I click "checkIV"

File: app/src/main/java/com/kamron/pogoiv/Pokefly.java [hunk 2]
```diff
@@ -991,9 +990,9 @@ private Pokemon interpretWhichPokemonUserInput() {
      * screenshot, and then deletes the screenshot.
      */
     private void deleteScreenShotIfRequired() {
-        if (batterySaver && !ssFile.isEmpty()) {
```
Comment (Blaisorblade) [inline app/src/main/java/com/kamron/pogoiv/Pokefly.java]: @sarav This is the only call to `isEmpty` and gets removed here. So I'm not sure about your message—and if it's something else that is crashing in `deleteScreenShotIfRequired`, I'm not sure what.

> I'm still getting a crash with Google Now on Tap share on isEmpty() in deleteScreenShotIfRequired() when I click "checkIV"

**Comment by Blaisorblade** [comment]
@sarav Happy to look at a stacktrace. Re Optional, see

https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained#optional

**Comment by sarav** [comment]
This is what happens when I try to test while falling asleep! I had build the apk, rebased to master and then pushed. This fix works. My bad.

**Pull request merged**